homeassistant/components/climate/nuheat.py

```diff
@@ -0,0 +1,212 @@
+class NuHeatThermostat(ClimateDevice):
+    """Representation of a NuHeat Thermostat."""
+    def __init__(self, api, serial_number, temperature_unit):
+        self._thermostat = api.get_thermostat(serial_number)
+        self._temperature_unit = temperature_unit
+        self._force_update = False
+
+    @property
+    def name(self):
+        """Return the name of the thermostat."""
+        return self._thermostat.room
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement."""
+        if self._temperature_unit == "C":
+            return TEMP_CELSIUS
+
+        return TEMP_FAHRENHEIT
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        if self._temperature_unit == "C":
+            return self._thermostat.celsius
+
+        return self._thermostat.fahrenheit
+
+    @property
+    def current_operation(self):
+        """Return current operation. ie. heat, idle."""
+        if self._thermostat.heating:
+            return STATE_HEAT
+
+        return STATE_IDLE
+
+    @property
+    def min_temp(self):
+        """Return the minimum supported temperature for the thermostat."""
+        if self._temperature_unit == "C":
+            return self._thermostat.min_celsius
+
+        return self._thermostat.min_fahrenheit
+
+    @property
+    def max_temp(self):
+        """Return the maximum supported temperature for the thermostat."""
+        if self._temperature_unit == "C":
+            return self._thermostat.max_celsius
+
+        return self._thermostat.max_fahrenheit
+
+    @property
+    def target_temperature(self):
+        """Return the currently programmed temperature."""
+        if self._temperature_unit == "C":
+            return self._thermostat.target_celsius
+
+        return self._thermostat.target_fahrenheit
+
+    @property
+    def target_temperature_low(self):
+        """Return the lower bound temperature we try to reach."""
+        return self.target_temperature
+
+    @property
+    def target_temperature_high(self):
+        """Return the upper bound temperature we try to reach."""
+        return self.target_temperature
+
+    @property
+    def current_hold_mode(self):
+        """Return current hold mode."""
+        if self.is_away_mode_on:
+            return MODE_AWAY
+
+        schedule_mode = self._thermostat.schedule_mode
+        if schedule_mode == SCHEDULE_RUN:
+            return MODE_AUTO
+
+        if schedule_mode == SCHEDULE_HOLD:
+            return MODE_HOLD_TEMPERATURE
+
+        if schedule_mode == SCHEDULE_TEMPORARY_HOLD:
+            return MODE_TEMPORARY_HOLD
+
+        return MODE_AUTO
+
+    @property
+    def operation_list(self):
+        """Return list of possible operation modes."""
+        return OPERATION_LIST
+
+    @property
+    def is_away_mode_on(self):
+        """
+        Return true if away mode is on.
+
+        Away mode is determined by setting and HOLDing the target temperature
```

Comment:
I'm not sure if that is useful. As far as I can see, most other platforms that support an away mode have hardware support for an away mode (i.e., the device itself has a different preset for "away" or whatever). I don't think we should hard-code the decision what temperature should be set in away mode here.

If a user wants to reduce the temperature when he's away, he can still use an automation to set the temperature to whatever value he desires, right?

Reply:
In my testing, having a service that easily toggles away mode has been useful and intuitive in both, the UI and in automations... Being able to just toggle away mode definitely makes more sense to my wife than setting/holding a low temperature.

However, I do agree that you should be able to configure your own minimum away temperature. I'll add support for that.

Reply:
I still think this is redundant. Why do you need this implemented inside the thermostat? Just have an `input_boolean` which, when turned on, calls `set_temperature` (with the wanted "away temperature") and `set_hold_mode` with the `temperature` hold mode.

This has the advantage that everyone can set his or her own desired "away temperature" instead of hard-coding it.

Reply:
Ah, I see that you have `min_away_temp` now. But that's not settable via the UI, is it? How should the user set it?

Also, I still dislike the fact that `away` is represented by setting the `away_mode` **and** the `hold_mode` at the same time. Storing states redundantly is in my opinion always a risk of getting into an inconsistent state. For example, consider this:

* I call `set_temperature` with whatever the away temperature is
* I call `set_hold_mode` with `temperature`

What will happen is that the hold mode magically changes to `away` and the away mode turns on, without me ever selecting away mode. I even explicitly selected a different hold mode!

Reply:
This is just providing a simple API to add away functionality to the thermostat. It's a missing feature that nearly every other smart thermostat has and is easily achieved with this component. NuHeat actually added a "works with nest" integration purely to add away mode to the thermostat by leveraging Nest's geofencing, etc. It's a feature that they wanted, but couldn't quite achieve.

It's easily solvable with APIs and the power of HASS... and really, it's the entire reason I built the component, heh.

Your example is pretty edge-casey. You'd have to deliberately set your temperature to whatever the minimum temperature is that the thermostat supports, which is 41F or 5C. But this is effectively what the existing "works with nest" integration does anyway.

Again, they're simply more coherent, shortcut APIs. People can choose which parts to use or not.

But anyway... this is getting a little pedantic.

Here's how I'm currently using it.

config.py
```
nuheat:
  username: !secret nuheat_username
  password: !secret nuheat_password
  devices: !secret nuheat_thermostat
  min_away_temp: 60
```

automations
```
- id: floor_off_when_leaving
  alias: Turn off heated floor when leaving
  trigger:
  - platform: state
    entity_id: group.family
    from: home
    to: not_home
  action:
    service: climate.set_away_mode
    data:
      entity_id: climate.master_bath
      away_mode: true

- id: floor_resume_when_arriving
  alias: Resume heated floor schedule when arriving
  trigger:
  - platform: state
    entity_id: group.family
    from: not_home
    to: home
  action:
    service: climate.set_away_mode
    data:
      entity_id: climate.master_bath
      away_mode: false
```

Here's a screenshot of what this looks like in Hass over time... (i haven't been away enough time for home to cool enough for it to be in use, heh)

![screen shot 2017-11-14 at 8 04 50 pm](https://user-images.githubusercontent.com/120263/32929571-8872852a-cb1e-11e7-82d5-f809c67b8d71.png)

Here's a screenshot of the thermostat in auto mode:

![screenshot_20171114-112212](https://user-images.githubusercontent.com/120263/32929684-5d59ceba-cb1f-11e7-89a7-92d6e271ec5d.png)

And again in away mode (which you'd probably never actually see):

![screenshot_20171114-095759](https://user-images.githubusercontent.com/120263/32929686-66e56e76-cb1f-11e7-8800-3b2c3503d152.png)

Reply:
What happens if:
* the thermostat is in `temperature` hold mode, say at 80°
* I activate away mode
* I deactivate away mode

My guess is that the thermostat goes into `home` hold mode? That's probably what `resume_schedule` does, right? And to me, that would be unexpected. I would assume that after I turn the away mode off, the thermostat returns to whatever setting I had before I activated the away mode.

Reply:
This seems consistent with what I would expect to happen.  

Deactivating away_mode should put the device back into its regular schedule.  I absolutely would not want the heater to crank back up just because I came home.

Reply:
If you wanted it to return to a specific temperature upon your return, you could, of course, do all of this with `set_temperature` and more complex automations.